internal/services/cdn/cdn_frontdoor_helpers.go

```diff
@@ -339,76 +383,265 @@ func updateRouteAssociations(d *pluginsdk.ResourceData, meta interface{}, routeI
 		updateProps.LinkToDefaultDomain = cdn.LinkToDefaultDomainEnabled
 	}
 
-	updatePrarams := azuresdkhacks.RouteUpdateParameters{
+	updateParams := azuresdkhacks.RouteUpdateParameters{
 		RouteUpdatePropertiesParameters: &updateProps,
 	}
 
-	future, err := workaroundsClient.Update(ctx, routeId.ResourceGroup, routeId.ProfileName, routeId.AfdEndpointName, routeId.RouteName, updatePrarams)
+	future, err := workaroundsClient.Update(ctx, routeId.ResourceGroup, routeId.ProfileName, routeId.AfdEndpointName, routeId.RouteName, updateParams)
 	if err != nil {
 		return fmt.Errorf("%s: updating the association with %s: %+v", *customDomainID, *routeId, err)
 	}
+
 	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
 		return fmt.Errorf("%s: waiting to update the association with %s: %+v", *customDomainID, *routeId, err)
 	}
 
 	return nil
 }
 
-func validateCustomDomanLinkToDefaultDomainState(resourceCustomDomains []interface{}, routeCustomDomains []interface{}, routeName string, routeProfile string) error {
-	// Make all of the custom domains belong to the same profile as the route...
+func validateCustomDomainLinkToDefaultDomainState(resourceCustomDomains []interface{}, routeCustomDomains []interface{}, routeName string, routeProfile string) error {
+	// NOTE: Only used in the deprecated custom domain link to default domain resource
+	if !features.FourPointOhBeta() {
+		// Make all of the custom domains belong to the same profile as the route...
+		wrongProfile := make([]string, 0)
+
+		for _, v := range resourceCustomDomains {
+			customDomain, err := parse.FrontDoorCustomDomainIDInsensitively(v.(string))
+			if err != nil {
+				return err
+			}
+
+			if customDomain.ProfileName != routeProfile {
+				wrongProfile = append(wrongProfile, fmt.Sprintf("%q", customDomain.ID()))
+			}
+		}
+
+		if len(wrongProfile) > 0 {
+			return fmt.Errorf("the following CDN FrontDoor Custom Domain(s) do not belong to the expected CDN FrontDoor Profile(Name: %q). Please remove the following CDN FrontDoor Custom Domain(s) from your CDN Route Disable Link To Default Domain configuration: %s", routeProfile, strings.Join(wrongProfile, ", "))
+		}
+
+		// Make sure the resource is referencing all of the custom domains that are associated with the route...
+		missingDomains := make([]string, 0)
+
+		for _, v := range routeCustomDomains {
+			// If this was updated by the portal, it lowercases to resource ID...
+			customDomain, err := parse.FrontDoorCustomDomainIDInsensitively(v.(string))
```

Comment:
this is validation for the Expand function, so we'll be validating these/can handle the API response in the Read instead so that these are consistent?

Reply:
Fixed.